source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -595,6 +601,33 @@ std::vector<Ssl::PrivateKeyMethodProviderSharedPtr> ContextImpl::getPrivateKeyMe
   return providers;
 }
 
+void ContextImpl::updateCertStats(Stats::Store& store) {
+  // Add CA cert expiration time.
+  std::vector<Ssl::CertificateNameExpiration> certs_expiration_time =
+      cert_validator_->getExpirationOfCerts();
+
+  // Add TLS certs' expiration time.
+  for (const auto& ctx : tls_contexts_) {
+    auto seconds_until_expiration =
+        Utility::getSecondsUntilExpiration(ctx.cert_chain_.get(), time_source_);
+    if (seconds_until_expiration.has_value()) {
+      certs_expiration_time.emplace_back(Ssl::CertificateNameExpiration{
+          ctx.cert_name_, std::chrono::duration<uint64_t>(seconds_until_expiration.value())});
+    }
+  }
+
+  for (const auto& cert : certs_expiration_time) {
+    const std::string& cert_name = cert.name;
+    uint64_t expiration_date = cert.expiration.count();
+    if (cert_stats_map_.find(cert_name) == cert_stats_map_.end()) {
```

Comment:
drive-by perf nit: use map::insert and capture the return pair, to save a hash lookup.
```
auto insertion = cert_stats_map_.insert(cert_name);
if (insertion.second) {
  insertion.first.second = new value
}
```
this would replace the `find` and the `operator[]` both of which will do the same hash lookup.

Reply:
Thanks. I used above approach in `source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc`.